sample/src/main/java/com/canhub/cropper/sample/camera_java/app/SCameraFragmentJava.java

```diff
@@ -0,0 +1,338 @@
+                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoUri);
+                startActivityForResult(takePictureIntent, CODE_PHOTO_CAMERA);
+            }
+        } catch (ActivityNotFoundException e) {
+            // display error state to the user
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void cameraPermissionLaunch() {
+        requestPermissionLauncher.launch(Manifest.permission.CAMERA);
+    }
+
+    @Override
+    public void showDialog() {
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
+        alertDialogBuilder.setTitle(R.string.missing_camera_permission_title);
+        alertDialogBuilder.setMessage(R.string.missing_camera_permission_body);
+                alertDialogBuilder.setPositiveButton(R.string.ok,
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface arg0, int arg1) {
+                                presenter.onOk();
+                            }
+                        });
+        alertDialogBuilder.setNegativeButton(R.string.cancel,new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                presenter.onCancel();
+            }
+        });
+        AlertDialog alertDialog = alertDialogBuilder.create();
+        alertDialog.show();
+
+    }
+
+    @Override
+    public void handleCropImageResult(@NotNull String uri) {
+        SCropResultActivity.Companion.start(this, null, Uri.parse(uri), null);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        presenter.onActivityResult(resultCode, requestCode, data);
+    }
+
+    private File createImageFile() throws IOException {
+        assert getActivity()!=null;
+        SimpleDateFormat timeStamp = new SimpleDateFormat(DATE_FORMAT, Locale.getDefault());
+        File storageDir = getActivity().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+        return File.createTempFile(
+                "$FILE_NAMING_PREFIX$$FILE_NAMING_SUFFIX",
+                FILE_FORMAT,
+                storageDir
+        );
+    }
+
+    public static class ViewHolder {
+        public final Button startWithUri, startWithoutUri, startPickImageActivity, startActivityForResult;
+
+        public ViewHolder(View view) {
+            startWithUri = view.findViewById(R.id.startWithUri);
+            startWithoutUri = view.findViewById(R.id.startWithoutUri);
+            startPickImageActivity = view.findViewById(R.id.startPickImageActivity);
+            startActivityForResult = view.findViewById(R.id.startActivityForResult);
+        }
+    }
```

Comment:
Can we use viewBinding? more safer for null cases, so we can focus on the library components